File: services/trader-dashboard/src/types/symbol.ts
import { AssetType } from "./asset-types";

// Symbol type definitions
export interface Symbol {
  symbol: string;
  baseSymbol: string;
  name: string;
  exchange: string;
  assetType: AssetType;
  isActive: boolean;
  lastPrice?: number;
  lastUpdate?: string;
}

export interface SymbolData {
  symbol: string;
  price: number;
  change: number;
  changePercent: number;
  volume: number;
  high: number;
  low: number;
  open: number;
  lastUpdate: string;
}

File: services/portfolio-service/src/main/java/com/quantis/market_data/model/Symbol.java
package com.quantis.market_data.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Symbol Model
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Symbol {
    private String symbol;
    private String baseSymbol;
    private String name;
    private String exchange;
    private AssetType assetType;
    private boolean isActive;
    private Double lastPrice;
    private String lastUpdate;
}
